DOCS-9458-email-send-editorial-kt' (#1142)
mulesoft/docs-connectors: email/1.3/modules/ROOT/edits/email-send.adoc

= Send Emails with Email Connector Examples - Mule 4
:keywords: email, connector, configuration, smtp, send, smtps
:page-aliases: connectors::email/email-send.adoc

> Anypoint Connector for Email (Email Connector) can send messages over SMTP and SMTPS servers by using the *Send* operation. The following examples show how to configure Email Connector to send emails and attachments, in both Anypoint Studio and the XML editor:

Reviewer: Minor point, but I wonder if we should reword this to put the audience into the action:
```suggestion
Use Anypoint Connector for Email (Email Connector) to send messages over SMTP and SMTPS servers by using the *Send* operation. The following examples show how to configure Email Connector to send emails and attachments, in both Anypoint Studio and the XML editor:
```

* <<send-attachments,Send Multiple Attachments>> +
Configure *Set Variable* components to define multiple media type attachments, and send these attachments using the Email Connector *Send* operation. Then log the result message using a *Logger* component.

[[send-emails]]
== Send Emails

The following example illustrates how to send emails over the SMTPS server. The flow initiates with a *Scheduler* component, and then the *Send* operation sends the email.
The SMTPS connection type enables SSL or TLS encryption and sends encrypted messages over the secured version of the SMTP server.

The following screenshot shows the app flow for this example:

.Email Connector Send messages over SMTPS
image::email-send-flow.png[Email Connector Send messages over SMTPS flow]

To create the flow: +

. Create a new Mule project in Studio.
. In the *Mule Palette* view, select the *Scheduler* component and drag it onto the canvas. +
This source initiates a flow when a time-based condition is met.
. Drag the Email *Send* operation to the right of the *Scheduler* source. +

> . On the *Send* configuration screen, click the plus sign (*+*) next to the *Connector configuration* field to configure a global element for the operation.

Reviewer: "on the screen"?
GLOBAL edit in other files in this set, re. "on" vs. "in"

Reviewer: Actually, I am going to review this document and add what I think is relevant to our style ref, so stand by....:
https://developers.google.com/style/ui-elements

. In the *Connection* field, select *SMTPS Connection*.
. In the *General* tab, enter the following values:
+
* *Host* +
`pop.gmail.com`
* *Port* +
`995`
* *User* +
`user3@domain.com`
* *Password* +
`passwordconnection`
+
[start=7]
. In the *TLS Configuration* section, select *Edit inline*.
. In the *Trust Store Configuration* section, enter the following values:
+
* *Path* +
`aTruststore.jks`
* *Password* +
`passwordtrust`
+
[start=9]
. In the *Key Store Configuration* section, enter the following values:
+
* *Path* +
`aKeystore`
* *Password*
`passwordkey`
+
[start=10]
. In the *Advanced* section, set *Enabled Protocols* to `TLSv1.2,SSLv3`.
. Click *OK* to close the configuration window.

The following screenshot shows the *Email SMTPS* configuration:

.Email SMTPS configuration
image::email-smtps-configuration.png[Email SMTPS configuration]

[start=12]

> . In the *Send* operation configuration screen, set the following values:
> +
> * *From address* +
> `user1@domain.com`
> * *To addresses* +
> `Edit inline`

Reviewer: I'm still thinking about this formatting, and I am asking for input....

The style ref. recommends inserting a blank line space between a list item and its description, which is perfectly fine in most cases and also helps with scanning and thinking about the elements as list item + separate paragraph.
But in lists like this one, that recommendation results in a lot of space:

```suggestion
. In the *Send* operation configuration screen, set the following values:
+
* *From address* +
+
`user1@domain.com`
* *To addresses* +
+
`Edit inline`
```
I hate to break my own rules, but when the description IS THIS SHORT, I think we might actually want to use this format (which I would never recommend in any other situation):

```suggestion
. In the *Send* operation configuration screen, set the following values:
+
* *From address*: `user1@domain.com`
* *To addresses*: `Edit inline`
```
What do YOU think?

Reviewer: Sorry! Back to COPYEDITING ONLY....

+
[start=13]
. Click the plus sign (*+*) to add new emails.
. Set the *Value* field to `user2@domain.com`.

> . Repeat the previous two actions and add another email as `user3@domain.com`.

Reviewer: ```suggestion
. Repeat the previous two actions but set the value to `user3@domain.com`.
```

. Set the *Subject* field to `IMPORTANT!`.
. Set the *Content* field to `"<h1>Hello this is an important message</h1>"` and the *ContentType* field to `text/html`. +

> If no body *Content* is specified, the content of the message payload is used by default. If that payload is not `text`, the operation fails with an `EMAIL:SEND` error.

Reviewer: ```suggestion
If *Content* is not empty, the content of the message payload is used by default. If that payload is not `text`, the operation fails with an `EMAIL:SEND` error.
```
OR
```suggestion
If no body content is specified, the content of the message payload is used by default. If that payload is not `text`, the operation fails with an `EMAIL:SEND` error.
```

http://www.mulesoft.org/schema/mule/tls http://www.mulesoft.org/schema/mule/tls/current/mule-tls.xsd">
	<email:smtp-config name="Email_SMTP" >
		<email:smtps-connection host="pop.gmail.com" port="995" user="user1@domain.com" password="passwordvalue" >
			<tls:context enabledProtocols="TLSv1.2,SSLv3" >
				<tls:trust-store path="aTruststore.jks" password="changeit" />
				<tls:key-store path="aKeystore" password="password" />
			</tls:context>
		</email:smtps-connection>
	</email:smtp-config>
	<flow name="emailconnectorFlow">
		<scheduler>
			<scheduling-strategy >
				<fixed-frequency />
			</scheduling-strategy>
		</scheduler>
		<email:send config-ref="Email_SMTP" fromAddress="user1@domain.com" subject="IMPORTANT!">
			<email:to-addresses >
				<email:to-address value="user3@domain.com" />
        <email:to-address value="user2@domain.com" />
			</email:to-addresses>
			<email:body contentType="text/html" >
				<email:content ><![CDATA["<h1>Hello this is an important message</h1>"]]></email:content>
			</email:body>
		</email:send>
	</flow>
</mule>
----

[[send-attachments]]
== Send an Attachment

> The following example illustrates how to send emails and attachments over the SMTP server. Use DataWeave to handle the attachments. The flow reads a JSON file using the File Connector *Read* operation and then uses the Email Connector *Send* operation to send the content of the file as an attachment:

Reviewer: "handle" > "process"?
I typically recommend not using "handle" unless within a context that mentions "handlers." Otherwise, the term can be temporarily misleading to nonnative English speakers.

The following screenshot shows the Anypoint Studio app flow for this example:

.Email Connector Send attachments over SMTP
image::email-attachment-flow.png[Email Connector Send attachments over SMTP]

To create the flow: +

. Create a new Mule project in Studio.

> . In the *Mule Palette* view, select the *Scheduler* component and drag it on to the canvas. +

Reviewer: ```suggestion
. In the *Mule Palette* view, select the *Scheduler* component and drag it onto the canvas. +
```

The source initiates a flow when a time-based condition is met.
. Drag the File Connector *Read* operation to the right of the *Scheduler* component.
. Set the *File Path* field to `file.json`.
. Drag the *Send* operation to the right of the *Read* operation. +
. On the *Send* configuration screen, click the plus sign (*+*) next to the *Connector configuration* field to configure a global element for the operation.
. In the *Connection* field, select *SMTP Connection*.
. In the *General* tab, enter the following values:

+
* *Host* +
`pop.gmail.com`
* *Port* +
`995`
* *User* +
`user1@domain.com`
* *Password* +
`password`
+
[start=9]
. Click *OK*.
. In the *Send* operation configuration screen, set the *To addresses* field to `Edit inline`.

> . Click the plus sign (*+*) to add a new email *Value* as `example@domain.com`.

Reviewer: ```suggestion
. Click the plus sign (*+*) to add a new email value, `example@domain.com`.
```

http://www.mulesoft.org/schema/mule/file http://www.mulesoft.org/schema/mule/file/current/mule-file.xsd
http://www.mulesoft.org/schema/mule/email http://www.mulesoft.org/schema/mule/email/current/mule-email.xsd">
	<email:smtp-config name="Email_SMTP">
		<email:smtp-connection host="pop.gmail.com" port="995" user="user1@domain.com" password="password" />
	</email:smtp-config>
	<flow name="emailattachment">
		<scheduler doc:name="Scheduler">
			<scheduling-strategy >
				<fixed-frequency />
			</scheduling-strategy>
		</scheduler>
		<file:read path="/file.json"/>
		<email:send config-ref="config" subject="Attachment test">
            <email:to-addresses >
                <email:to-address value="example@domain.com" />
            </email:to-addresses>
            <email:body >
                <email:content ><![CDATA[#["<h1>Hello this is an important message</h1>"]]]></email:content>
            </email:body>
            <email:attachments >#[{
          'json-attachment' : "payload"
        }]</email:attachments>
        </email:send>
	</flow>
</mule>
----

== Send Multiple Attachments

> The following example illustrates how to send emails and multiple attachments over the SMTP server. The flow initiates with a *Scheduler* component. Then, *Set Variable* components define each attachment media type JSON, text and file. Subsequently, the Email Connector *Send* operation sends the attachments in the email. Then log the result message using a *Logger* component.

Reviewer: ```suggestion
The following example illustrates how to send emails and multiple attachments over the SMTP server. The flow initiates with a *Scheduler* component. Then, *Set Variable* components define each attachment media type JSON, text and file. Subsequently, the Email Connector *Send* operation sends the attachments in the email, after which you log the result message using a *Logger* component.
```

The following screenshot shows the Anypoint Studio app flow for this example:

.Email Connector Multiple attachments
image::email-multipleattachment-flow.png[Email Connector Multiple attachments]

To create the flow: +

. Create a new Mule project in Studio.

> . In the *Mule Palette* view, select the *Scheduler* component and drag it on to the canvas. +

Reviewer: ```suggestion
. In the *Mule Palette* view, select the *Scheduler* component and drag it onto the canvas. +
```
